packages/class/src/bmi.ts

```diff
@@ -0,0 +1,137 @@
+import { CLASS } from "./class";
+import { classConfig, ClassConfig } from "./config";
+
+/**
+ * A lightweight [BMI](https://bmi.readthedocs.io) like interface for the CLASS model.
+ *
+ * Inspiration https://github.com/uihilab/BMI-JS/blob/main/bmijs/bmi.js
+ *
+ * Deviations from the BMI standard
+ * - accessors do not use dest argument
+ * - initialize() accepts object instead of string
+ * - parameters() returns default config
+ */
+interface BmiLight<Config> {
+  initialize(config: Config): void;
+  update(): void;
+  get_component_name(): string;
+  get_output_item_count(): number;
+  get_output_var_names(): string[];
+  get_var_grid(name: string): number;
+  get_var_type(name: string): string;
+  get_var_location(name: string): string;
+  get_current_time(): number;
+  get_end_time(): number;
+  get_time_units(): string;
+  get_time_step(): number;
+  get_value(name: string): number[];
+  get_grid_type(): string;
+}
+
+const ouput_var_names: string[] = ["h", "theta", "dtheta", "q", "dq"] as const;
+
+export class BmiClass implements BmiLight<ClassConfig> {
+  config: ClassConfig = classConfig.parse({});
+  model: CLASS = new CLASS(this.config);
+
+  constructor() {
+    // no-op
+  }
+
+  // Model control functions
+
+  initialize(config: ClassConfig) {
+    this.config = config;
+    this.model = new CLASS(config);
+  }
+
+  update() {
+    this.model.update();
+  }
+
+  // Model information functions
+
+  get_component_name() {
+    return "Chemistry Land-surface Atmosphere Soil Slab model";
+  }
+
+  get_output_item_count() {
+    return ouput_var_names.length;
+  }
+
+  get_output_var_names() {
+    return ouput_var_names;
+  }
+
+  // Variable information functions
+
+  get_var_grid(name: string) {
+    return 1;
+  }
+
+  get_var_type(name: string) {
+    return "float";
+  }
+
+  get_var_location(name: string) {
+    return "node";
+  }
+
+  // Time functions
+  get_current_time() {
+    return this.model.t;
+  }
+
+  get_end_time() {
+    return this.config.timeControl.runtime;
+  }
+
+  get_time_units() {
+    return "s";
+  }
+
+  get_time_step() {
+    return 1;
+  }
+
+  // Variable getter and setter functions
+
+  get_value(name: string) {
+    if (ouput_var_names.includes(name)) {
+      return [this.model[name as keyof CLASS]] as [number];
+    }
+    throw new Error(`Variable ${name} not found`);
+  }
+
+  // Model grid functions
+
+  get_grid_type() {
+    return "scalar";
+  }
+
+  // Extra methods
+
+  run<T extends string[]>({
```

Comment:
This could use some cleanup; perhaps at least add a docstring for now, since it's also not a standard BMI function?

Reply:
docstring added